Markdown/Markdown.sublime-syntax

```diff
@@ -219,6 +219,18 @@ contexts:
     - match: '^[ ]{0,3}(?=(?:{{html_tag_open_commonmark}}|{{html_tag_close_commonmark}})\s*$|<{{html_tag_block_end_at_blank_line}})'
       comment: Markdown formatting is disabled inside block level tags and if a complete HTML tag is the only thing on the line.
       push: [disabled-markdown-pop-at-eol, disable-markdown-pop-at-blank-line]
+    - match: |-
+        (?x:
+            \s*                        # Leading whitespace
+            (\[)\^(.+?)(\])(:)         # Reference name
+            (.*)                       # The footnote
```

Comment:
it looks like footnotes don't have to end at eol, and can span multiple lines and contain block-level elements: https://michelf.ca/projects/php-markdown/extra/#footnotes
also, the `(.+?)` should probably also be changed to match a valid HTML `id` attribute as per the same docs